test/src/tests-user.ts

```diff
@@ -99,25 +81,32 @@ describe('mParticle User', function() {
             ).to.have.property('Timestamp', 10);
             done();
         });
+        });
 
         it('get/set consent state for multiple users', done => {
             mParticle._resetForTests(MPConfig);
 
             mParticle.init(apiKey, mParticle.config);
 
-            mockServer.respondWith(urls.login, [
-                200,
-                {},
-                JSON.stringify({ mpid: 'MPID1', is_logged_in: false }),
-            ]);
-
+            waitForCondition(hasIdentifyReturned)
+            .then(() => {
             const userIdentities1 = {
                 userIdentities: {
                     customerid: 'foo1',
                 },
             };
 
+            fetchMockSuccess(urls.login, {
+                mpid: 'loginMPID1', is_logged_in: false
+            });
+
             mParticle.Identity.login(userIdentities1);
+            waitForCondition(() => {
+                return (
+                    mParticle.Identity.getCurrentUser()?.getMPID() === 'loginMPID1'
```

Comment:
I think you can reuse my `hasLoginReturned` helper.

Reply:
similar to your other comment, they are not in utils.  all instances of `waitForCondition` should be updated when we go through this ticket https://mparticle-eng.atlassian.net/browse/SQDSDKS-6849